### PR TITLE
improvement(marginfi-v2-ui): add Coinbase and SquadsX wallets to extended options 

### PR DESCRIPTION
**Pull request by masterzorgon**
## Issue
Both the Coinbase and SquadsX team raised issues with me this past week that users are unable to use their respective wallets to onboard to marginfi. 

## Solution
I add SquadsX and Coinbase wallet adapters inside the `src/hooks/useAvailableWallet.ts` file and conditionally added these adapters to the `addAbsentWallets` function.

### CODE REVIEW

**Review by k0beLeenders — CHANGES_REQUESTED** [review]
Can you rebase this branch on main @masterzorgon

**Comment by masterzorgon** [comment]
> Can you rebase this branch on main @masterzorgon

rebased!

**Review by k0beLeenders — CHANGES_REQUESTED** [review]
I see a few unrelated changes, something seems to have gone wrong with the rebase

File: apps/marginfi-v2-ui/src/utils/mrgnActions.ts [hunk 1]
```diff
@@ -879,6 +879,7 @@ export async function mintLstToken({
 // Helpers //
 // ------------------------------------------------------------------//
 
+// PDA not being able to create a bank?
```
Comment (k0beLeenders) [inline apps/marginfi-v2-ui/src/utils/mrgnActions.ts:882]: what do you mean with this?

**Comment by k0beLeenders** [comment]
I made some changes to the onboarding process. Previously, we filtered the wallet list to show only relevant options for social, Sol, and Eth flows. However, this caused an issue where users could not find their wallet during onboarding. To address this, I removed the forced filter for the Sol and Eth flows so that all installed wallets are now visible. Additionally, I have reordered the wallets to prioritize certain ones. 

let me know your thoughts @masterzorgon @chambaz

**Review by k0beLeenders — APPROVED** [review]

File: apps/marginfi-v2-ui/src/components/common/Wallet/components/AuthenticationV2/AuthDialog.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ export const AuthDialog = () => {
     return isOnboarded !== null ? "RETURNING_USER" : "ONBOARD_MAIN";
   }, []);
 
-  const [flow, setFlow] = React.useState<AuthFlowType>(mainFlow);
+  const [flow, setFlow] = React.useState<AuthFlowType>("ONBOARD_ETH");
```
[the remark this answers is not in the document]
Comment (chambaz) [reply apps/marginfi-v2-ui/src/components/common/Wallet/components/AuthenticationV2/AuthDialog.tsx]: woah, close one. lets make sure to fully test everything :)

**Comment by chambaz** [comment]
> I made some changes to the onboarding process. Previously, we filtered the wallet list to show only relevant options for social, Sol, and Eth flows. However, this caused an issue where users could not find their wallet during onboarding. To address this, I removed the forced filter for the Sol and Eth flows so that all installed wallets are now visible. Additionally, I have reordered the wallets to prioritize certain ones.
> 
> let me know your thoughts @masterzorgon @chambaz

I tend to agree here. I think the way we had attempted to do it makes sense, reducing choice paralysis, and simplifying the flow. however from feedback users and wallet providers expect to see their option available and I think your latest solution is the easiest option. one day I pray for a true Wallet Standard and this implementation nightmare goes away.

File: apps/marginfi-v2-ui/src/utils/mrgnActions.ts (hunk 1 above)
Comment (masterzorgon) [reply apps/marginfi-v2-ui/src/utils/mrgnActions.ts:882]: this was a comment i made while debugging a SquadsX issue. it was a note to myself and forgot to delete it. i was mistaken in thinking at our code wasn't allowing off-curve signers to create banks

**Comment by masterzorgon** [comment]
> I made some changes to the onboarding process. Previously, we filtered the wallet list to show only relevant options for social, Sol, and Eth flows. However, this caused an issue where users could not find their wallet during onboarding. To address this, I removed the forced filter for the Sol and Eth flows so that all installed wallets are now visible. Additionally, I have reordered the wallets to prioritize certain ones.
> 
> let me know your thoughts @masterzorgon @chambaz

**Comment by masterzorgon** [comment]
> I made some changes to the onboarding process. Previously, we filtered the wallet list to show only relevant options for social, Sol, and Eth flows. However, this caused an issue where users could not find their wallet during onboarding. To address this, I removed the forced filter for the Sol and Eth flows so that all installed wallets are now visible. Additionally, I have reordered the wallets to prioritize certain ones.
> 
> let me know your thoughts @masterzorgon @chambaz



> I made some changes to the onboarding process. Previously, we filtered the wallet list to show only relevant options for social, Sol, and Eth flows. However, this caused an issue where users could not find their wallet during onboarding. To address this, I removed the forced filter for the Sol and Eth flows so that all installed wallets are now visible. Additionally, I have reordered the wallets to prioritize certain ones.
> 
> let me know your thoughts @masterzorgon @chambaz

I see that the SquadsX Wallet option is not available during the onboarding flow.

Screenshot 2024-07-05 at 3 23 12 PM

**Comment by k0beLeenders** [comment]
Rebasing on main rn, but should be good to go when @chambaz approves it

**Pull request merged**